# HG changeset patch
# User Steffen Groot <steffen.groot@technolution.eu>
# Date 1235161427 18000
#      Fri Feb 20 15:23:47 2009 -0500
# Node ID b86b42f77899b4131713ec747fbd826bcf24d833
# Parent  9dddbad47122afb781ed30c14c88168aa63a0856
__patch__.m: fix indexing of varargin

diff --git a/doc/interpreter/contributors.in b/doc/interpreter/contributors.in
--- a/doc/interpreter/contributors.in
+++ b/doc/interpreter/contributors.in
@@ -54,16 +54,17 @@ Walter Gautschi
 Klaus Gebhardt
 Driss Ghaddab
 Nicolo Giorgetti
 Michael Goffioul
 Glenn Golden
 Tomislav Goles
 Keith Goodman
 Brian Gough
+Steffen Groot
 Etienne Grossmann
 Peter Gustafson
 Kai Habel
 William P. Y. Hadisoeseno
 Jaroslav Hajek
 Benjamin Hall
 Kim Hansen
 Soren Hauberg
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-02-20  Steffen Groot  <steffen.groot@technolution.eu>
+
+	* plot/__patch__.m: Correct indexing of varargin.
+
 2009-02-20  Jaroslav Hajek  <highegg@gmail.com>
 
 	* linear-algebra/dmult.m: Remove.
 	* linear-algebra/Makefile.in: Update.
 	* deprecated/dmult.m: Move here; revert to the 3.0.x version.
 	* deprecated/Makefile.in: Update.
 	* statistics/models/logistic_regression_derivatives.m: Replace dmult
 	by diagonal matrices.
diff --git a/scripts/plot/__patch__.m b/scripts/plot/__patch__.m
--- a/scripts/plot/__patch__.m
+++ b/scripts/plot/__patch__.m
@@ -64,17 +64,17 @@ function [h, fail] = __patch__ (p, varar
     endif
     
     if (strcmpi (varargin{1}, "faces"))
       faces = varargin{2};
       if (strcmpi (varargin{3}, "vertices"))
 	vert = varargin{4};
 	have_faces = true;
       endif
-    elseif (strcmpi (varargin{3}, "vertices"))
+    elseif (strcmpi (varargin{1}, "vertices"))
       vert = varargin{2};
       if (strcmpi (varargin{3}, "faces"))
 	faces = varargin{4};
 	have_faces = true;
       endif
     endif
     if (!have_faces)
       fail = true;
